Managers/FrameMan.cpp

```diff
@@ -1,2651 +1,1391 @@
-		// server will pick them to build the frame state and send to client
-		if (m_StoreNetworkBackBuffer)
-		{
-			int layerCount = 0;
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-			for (std::list<SceneLayer *>::reverse_iterator itr = g_SceneMan.GetScene()->GetBackLayers().rbegin(); itr != g_SceneMan.GetScene()->GetBackLayers().rend(); ++itr)
-			{
-				SLOffset[whichScreen][layerCount] = (*itr)->GetOffset();
-				layerCount++;
+	void FrameMan::FlipFrameBuffers() {
+		if ((!m_Fullscreen && m_NxWindowed != 1) || (m_Fullscreen && m_NxFullscreen != 1)) {
+			stretch_blit(m_pBackBuffer32, screen, 0, 0, m_pBackBuffer32->w, m_pBackBuffer32->h, 0, 0, SCREEN_W, SCREEN_H);
+		} else {
+			blit(m_pBackBuffer32, screen, 0, 0, 0, 0, m_pBackBuffer32->w, m_pBackBuffer32->h);
+		}
+	}
 
-				if (layerCount >= MAX_LAYERS_STORED_FOR_NETWORK)
-					break;
-			}
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void FrameMan::SetTransTable(TransparencyPreset transSetting) {
+		switch (transSetting) {
+			case LessTrans:
+				color_map = &m_LessTransTable;
+				break;
+			case MoreTrans:
+				color_map = &m_MoreTransTable;
+				break;
+			case HalfTrans:
+				color_map = &m_HalfTransTable;
+				break;
+			default:
+				RTEAbort("Undefined transparency preset value passed in. See TransparencyPreset enumeration for defined values.")
 		}
+	}
 
-        Vector targetPos = g_SceneMan.GetOffset(whichScreen);
-        // Adjust the drawing position on the target screen for if the target screen is larger than the scene in nonwrapping dimension.
-        // Scene needs to be displayed centered on the target bitmap then, and that has to be adjusted for when drawing to the screen
-        if (!g_SceneMan.SceneWrapsX() && pDrawScreen->w > g_SceneMan.GetSceneWidth())
-            targetPos.m_X += (pDrawScreen->w - g_SceneMan.GetSceneWidth()) / 2;
-        if (!g_SceneMan.SceneWrapsY() && pDrawScreen->h > g_SceneMan.GetSceneHeight())
-            targetPos.m_Y += (pDrawScreen->h - g_SceneMan.GetSceneHeight()) / 2;
-
-		// Try to move at the framebuffer copy time to maybe prevent wonkyness
-		m_TargetPos[m_NetworkFrameCurrent][whichScreen] = targetPos;
-
-        // Draw the scene
-		if (!m_StoreNetworkBackBuffer/* || g_UInputMan.KeyHeld(KEY_6)*/)
-		{
-			g_SceneMan.Draw(pDrawScreen, pDrawScreenGUI, targetPos);
-		} 
-		else 
-		{
-			clear_to_color(pDrawScreen, g_KeyColor);
-			clear_to_color(pDrawScreenGUI, g_KeyColor);
-			g_SceneMan.Draw(pDrawScreen, pDrawScreenGUI, targetPos, true, true);
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	int FrameMan::DrawLine(BITMAP *pBitmap, const Vector &start, const Vector &end, int color, int altColor, int skip, int skipStart, bool shortestWrap) {
```

Comment:
This isn't really working out because we end up with junk arguments and I don't like that.
It's ok when you use it to draw a non-dot line, but if you use it for dots you have to fill in the color arguments which are irrelevant.
Other than that, yeah, it was a nice simple merge.

`DrawDotLine(bitmap, start, end, color, altColor = 0, skip = 0, skipStart = 0, shortestWrap = false, drawDot = false, dot = 0);`

I tried to do it in a different order but then you end up having to fill all of it for both applications.
Plus having 10 arguments feels criminal.

So i'm just gonna keep these two separate but you can poke at it if you want.

Reply:
Hmm, I see what you're saying, but I also think you misunderstood me (but also it's been a while so maybe I'm misunderstanding). What I think I wanted here was:
```
int FrameMan::DrawLine(normalArgs) {
    return FrameMan::SharedLineDrawingMethodThing(normalArgs);
}
int FrameMan::DrawDotLine(normalDotArgs) {
    assert(dot isn't null);
    return FrameMan::SharedLineDrawingMethodThing(normalDotArgs and 0s for the 2 colours);
}
private int FrameMan::SharedLineDrawingMethodThing(sharedArgs, color, altColor, dotBitmap) {
    do all the shared code
}
```
That way the public draw methods happily stay the same and the extra ugliness gets hidden in a private method. It does mean that private method they both use will have a disgusting 9 arguments, but it also saves almost 100 lines of literally duplicated code so I think it's worth it. Also, technically since the dot argument is a pointer, I think you could do `method(sharedArgs, colorOrPointerToBitmapAsInt, altColor) {` but that's super gross so you shouldn't do it!

I guess also some of these arguments could be combined, instead of start and end as separate Vectors we could do `Vector[2] startAndEnd` or `std::pair<Vector, Vector> startAndEnd`. Same thing with color and altColor and skip and skipStart for that matter. I'm okay with or without this change, I don't like the huge number of arguments but changing it like this may just obfuscate the method.
Thinking about it, I guess another option and maybe the most elegant one would be to make a struct and pass that around.